### PR TITLE
[GW-7313] [GW-7334]

### PR DESCRIPTION
**Pull request by LuqmanHakim-Grune**
[GW-7313]
- Create a user-registration-order.ts model
- Add methods of: createUserRegistrationOrder()
- Register to models/index.js
- Adjust register process, create UserRegistrationOrder instead of User data
[GW-7334]
- Send email authentication with token when registration finished
- Add sendPasswordResetEmail
- Add TODO list for next work

### CODE REVIEW

File: packages/app/src/server/routes/login.js [hunk 1]
```diff
@@ -10,6 +11,7 @@ module.exports = function(crowi, app) {
   const logger = loggerFactory('growi:routes:login');
   const path = require('path');
   const User = crowi.model('User');
+  const UserRegistrationOrder = crowi.model('UserRegistrationOrder');
```
Comment (yuki-takei) [inline packages/app/src/server/routes/login.js:14]: Do not retrieve from `crowi.model`. Use import syntax.

File: packages/app/src/server/routes/login.js [hunk 2]
```diff
@@ -120,14 +122,9 @@ module.exports = function(crowi, app) {
           return res.redirect('/register');
         }
 
-        User.createUserByEmailAndPassword(name, username, email, password, undefined, async(err, userData) => {
+        UserRegistrationOrder.createUserRegistrationOrder(name, username, email, password, undefined, async(err, userData) => {
```
Comment (yuki-takei) [inline packages/app/src/server/routes/login.js:125]: Do not edit this method.
This is very old code that can be degraded and is risky.

File: packages/app/src/server/models/user-registration-order.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
+/* eslint-disable no-use-before-define */
```
Comment (yuki-takei) [inline packages/app/src/server/models/user-registration-order.js:1]: - Implement as a typescript file
    - Refs: update-post.ts
- Implement as a scratch code
    - Do not base it on the current User schema

**Review by yuki-takei — CHANGES_REQUESTED** [review]

File: packages/app/src/server/models/index.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ module.exports = {
   // PageArchive: require('./page-archive'),
   PageTagRelation: require('./page-tag-relation'),
   User: require('./user'),
+  UserRegistrationOrder: require('./user-registration-order'),
```
Comment (kaoritokashiki) [inline packages/app/src/server/models/index.js:7]: please do not set the UserRegistrationOrder model here since setting models here is an old method

**Review by kaoritokashiki — CHANGES_REQUESTED** [review]

File: packages/app/src/server/routes/forgot-password.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import {
 import { ReqWithPasswordResetOrder } from '../middlewares/inject-reset-order-by-token-middleware';
 
 export const forgotPassword = (req: Request, res: Response): void => {
+  // TODO: grune - make different view to bypass update password
```
Comment (kaoritokashiki) [inline packages/app/src/server/routes/forgot-password.ts]: please do not add code here for the same reason above.
create a new endpoint on the login.js file  instead

**Review by kaoritokashiki — CHANGES_REQUESTED** [review]

**Pull request closed without merging**